### PR TITLE
Create login and session

### PR DESCRIPTION
**Pull request by jcccookie**
I created a login api and session to be stored in browser and database.

### CODE REVIEW

File: routes/account.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
-const bcrypt = require('bcryptjs');
+const bcrypt = require('bcryptjs'); // Can't get the same result and verify the password
```
Comment (Jack-Edwards) [inline routes/account.js:1]: Should we just get rid of `bcrypt`?

File: routes/account.js [hunk 2]
```diff
@@ -47,6 +49,72 @@ module.exports = function(pool) {
...
+                httpOnly: true
+            })
+        }
+
+        return new Promise((resolve, reject) => {
+            pool.query(
+                'SELECT Password, SessionId FROM user WHERE Email = ?;', 
+                [email], 
+                (error, result) => {
+                    if (error) return reject(error)
```
Comment (Jack-Edwards) [inline routes/account.js:68]: Does `reject(error)` return the SQL error to the client/requestor?  This seems like a bad practice.

File: routes/account.js [hunk 3]
```diff
@@ -47,6 +49,72 @@ module.exports = function(pool) {
...
+        // Verify password
+        .then(({ account }) => {
+            const hashedPassword = hash(req.body.password) 
+
+            if (account && account.Password === hashedPassword) {
+                return { sessionId: account.SessionId }
+            } else {
+                const error = new Error('Incorrect username/password')            
+
+                throw error
```
Comment (Jack-Edwards) [inline routes/account.js:82]: Returning 'Incorrect username/password' to the client/requestor seems more appropriate than throwing an error and returning an HTTP 500 response.  Am I understanding this correct?

File: routes/account.js [hunk 4]
```diff
@@ -47,6 +49,72 @@ module.exports = function(pool) {
...
+                sessionString = `${email}|${sessionId}`
+
+                setSessionCookie({ sessionString, res })
+            } else {
+                const id = uuid()
+                sessionString = `${email}|${id}`
+
+                return new Promise((resolve, reject) => {
+                    pool.query(
+                        'UPDATE user SET SessionId = ? WHERE Email = ?;',
```
Comment (Jack-Edwards) [inline routes/account.js:100]: What do you think about doing `'UPDATE user SET SessionId = ? WHERE Id = ?;'` ?

File: routes/account.js (hunk 1 above)
Comment (jcccookie) [reply routes/account.js:1]: Yeah I'll remove it.

File: routes/account.js (hunk 2 above)
Comment (jcccookie) [reply routes/account.js:68]: No it does not. I'll follow the practice you mentioned in the last meeting

File: routes/account.js (hunk 4 above)
Comment (jcccookie) [reply routes/account.js:100]: I think that works better because ids are unique. I'll try to retrieve the id from the database

**Review by Jack-Edwards — APPROVED** [review]

**Pull request merged**